clients/wallet/src/stellar_wallet.rs

```diff
@@ -191,25 +224,52 @@ impl StellarWallet {
 
 	/// Submits transactions found in the wallet's cache to Stellar.
 	/// Returns a tuple: (list of txs successfully submitted, list of errors of txs not resubmitted)
-	pub async fn resubmit_transactions_from_cache(&self) -> (Vec<TransactionResponse>, Vec<Error>) {
+	pub async fn resubmit_transactions_from_cache(
+		&self,
+	) -> Vec<oneshot::Receiver<Result<TransactionResponse, Error>>> {
 		let _ = self.transaction_submission_lock.lock().await;
 		let horizon_client = Client::new();
 
-		let mut passed = vec![];
+		let mut return_val = vec![];
```

Comment:
Let's rename this. The doc comment above also needs to be updated I think.

Maybe to (?)
```suggestion
		let mut error_receivers = vec![];
```

Reply:
done